WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/widget/ViewsWidgetListProvider.kt

```diff
@@ -0,0 +1,78 @@
+package org.wordpress.android.ui.stats.refresh.lists.widget
+
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.widget.RemoteViews
+import android.widget.RemoteViewsService.RemoteViewsFactory
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.ui.stats.refresh.lists.widget.StatsViewsWidgetConfigureViewModel.Color
+import javax.inject.Inject
+
+class ViewsWidgetListProvider(val context: Context, intent: Intent) : RemoteViewsFactory {
+    @Inject lateinit var viewModel: ViewsListViewModel
+    private var showChangeColumn: Boolean = intent.getBooleanExtra(SHOW_CHANGE_VALUE_KEY, true)
+    private var colorModeId: Int = intent.getIntExtra(COLOR_MODE_KEY, Color.LIGHT.ordinal)
+    private var siteId: Long = intent.getLongExtra(SITE_ID_KEY, 0L)
+
+    init {
+        (context.applicationContext as WordPress).component().inject(this)
+    }
+
+    override fun onCreate() {
+        viewModel.start(siteId, colorModeId, showChangeColumn)
+    }
+
+    override fun getLoadingView(): RemoteViews? = null
+
+    override fun onDataSetChanged() {
+        viewModel.onDataSetChanged()
+    }
+
+    override fun hasStableIds(): Boolean = true
+
+    override fun getViewTypeCount(): Int = 1
+
+    override fun onDestroy() {
+    }
+
+    override fun getCount(): Int {
+        return viewModel.data.size
+    }
+
+    override fun getItemId(position: Int): Long {
+        return viewModel.data[position].key.hashCode().toLong()
+    }
+
+    override fun getViewAt(position: Int): RemoteViews {
+        val uiModel = viewModel.data[position]
+        val rv = RemoteViews(context.packageName, uiModel.layout)
+        rv.setTextViewText(R.id.period, uiModel.key)
+        if (uiModel.isNeutralChangeVisible) {
```

Comment:
I assume only one type of change could be visible at the time, so I wonder, maybe we could simplify this part by passing enum to `uiModel ` and use it (inside `when`, for example) to pick the view id?

Reply:
That's true but neither can be shown (if there is no previous item). This would mean we'd need some extra logic that's now in the ViewModel. I think I can create a new enum to handle these 4 usecases though 👍